test/configuration/test_cases/test_39_environment_modification.xml

```diff
@@ -0,0 +1,173 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DS xmlns="http://www.eprosima.com/XMLSchemas/discovery-server" user_shutdown="false">
+
+    <servers>
+        <server name="server_1" profile_name="UDP_server_1" creation_time="5" />
+        <server name="server_2" profile_name="UDP_server_2" creation_time="18" />
+    </servers>
+
+    <clients>
+        <client name="client1" profile_name="UDP_client1_server1" creation_time="5">
+            <publisher topic="topic1" />
+        </client>
+        <client name="client3" profile_name="UDP_client1_server2" creation_time="18">
+            <subscriber topic="topic1" />
+        </client>
+    </clients>
+
+    <simples>
+        <simple name="client2" profile_name="UDP_client2_server1" creation_time="5">
+            <subscriber topic="topic1" />
+        </simple>
+    </simples>
+
+
+    <snapshots file="./test_39_environment_modification~">
+        <snapshot time="8">test_39_environment_modification_initial</snapshot>
+        <snapshot time="12">test_39_environment_modification_add_server2</snapshot>
+        <snapshot time="16">test_39_environment_modification_add_server1</snapshot>
+        <snapshot time="22">test_39_environment_modification_final</snapshot>
+    </snapshots>
+
+    <environment>
+        <change time="3" key="ROS_DISCOVERY_SERVER">;;localhost:65000</change>
+        <change time="9" key="ROS_DISCOVERY_SERVER">;localhost:38812;localhost:65000</change>
+        <change time="13" key="ROS_DISCOVERY_SERVER">localhost:38811;localhost:38812;localhost:65000</change>
+    </environment>
```

Comment:
I think the test can be optimized and maybe including and explanation of the steps could also help understanding the test.

1. First, I suppose that waiting 5 seconds to create any participant is not optimal.
2. First the environment variable is modified to include the second server and client (t=9) that are not created until t=18. Maybe the order of the test should be rethought.
3. I understand that 2 clients are created in order to connect to its own server (the environment variable does not affect them). The simple participant is transformed into a client and it will discover the servers with the environment variable. Finally, when the environment variable allows one server to discover the other one, all participants discover each other.